src/safeposix/syscalls/sys_calls.rs

```diff
@@ -427,21 +460,29 @@ impl Cage {
                     Epoll(p) => p.flags & O_CLOEXEC,
                 } != 0
                 {
+                    // If the flag is set - we add the fd to our vector
                     cloexecvec.push(fd);
                 }
             }
         }
 
+        //For each fd in our close  vector list
+        //We call the close_syscall which takes in a file decsriptor
+        //and closes it
         for fdnum in cloexecvec {
             self.close_syscall(fdnum);
         }
 
-        // we grab the parent cages main threads sigset and store it at 0
-        // this way the child can initialize the sigset properly when it establishes its
-        // own mainthreadid
+        // We clone the parent cage's main threads and store them and index 0
```

Comment:
curious to know how you learnt about this behaviour :)

Reply:
Saw this behaviour during fork syscall - hence learnt it while documenting that